### PR TITLE
Add some operations

### PR DESCRIPTION
**Pull request by yangw1234**
## What changes were proposed in this pull request?

Add some operations:
AvgPool, AvgPoolGrad, BiasAddV1, SigmoidGrad, Relu6, Relu6Grad, Elu, EluGrad, Softplus, SoftplusGrad, LogSoftmax, Softsign, SoftsignGrad, Abs, LessEqual, GreaterEqual, ApproximateEqual, Log, LogGrad, Log1p, Log1pGrad, SquaredDifference, Div

## How was this patch tested?

add unit tests

### CODE REVIEW

**Comment by yangw1234** [comment]
pr validation pass

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Sigmoid.scala [hunk 1]
```diff
@@ -43,8 +43,15 @@ class Sigmoid[@specialized(Float, Double) T: ClassTag](
...
 
   override def updateGradInput(input: Tensor[T], gradOutput: Tensor[T]): Tensor[T] = {
-    gradInput.resizeAs(input).copy(gradOutput)
-    gradInput.cmul(output).cmul(buffer.add(ev.fromType(-1))).cmul(output)
+    updateGradInputInternal(output, gradOutput)
+  }
+
+  private[bigdl] def updateGradInputInternal(output: Tensor[T],
+                                             gradOutput: Tensor[T]): Tensor[T] = {
+    gradInput.resizeAs(gradOutput).copy(gradOutput)
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Sigmoid.scala]: why change to this format?

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/TensorflowSpecHelper.scala [hunk 2]
```diff
@@ -83,8 +83,15 @@ abstract class TensorflowSpecHelper extends BigDLSpecHelper {
       bigdlOutput.toTable.apply[Tensor[_]](outputIndex + 1)
     }
     val tfOutput = runGraphTF(graphFile, nodeDefBuilder.getName + s":$outputIndex")
-    bigdlOutputTensor.asInstanceOf[Tensor[NumericWildCard]]
-      .almostEqual(tfOutput.asInstanceOf[Tensor[NumericWildCard]], delta) should be(true)
+
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/TensorflowSpecHelper.scala]: there's another api. Please use it to get the tensor for your specific compare

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Sigmoid.scala (hunk 1 above)
Comment (yangw1234) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Sigmoid.scala]: The tensorflow sigmoid grad use dy/dx = y(1-y) to calculate the gradient. The SigmoidGrad op will use the `updateGradInputInternal` function

Comment (yiheng) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Sigmoid.scala]: the result is equal. I wonder why to change to y(1-y) from original

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/loaders/UnaryOpBase.scala [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.utils.tf.loaders
+
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.utils.tf.TensorflowSpecHelper
+import org.tensorflow.framework.{DataType, NodeDef}
+import com.intel.analytics.bigdl.utils.tf.Tensorflow._
+
+abstract class UnaryOpBase extends TensorflowSpecHelper {
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/loaders/UnaryOpBase.scala]: UnaryOpBaseSpec?

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/loaders/UnaryOpBase.scala [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  def getOpName: String
+
+  def getInput: Tensor[_]
+
+  s"$getOpName forward float" should "be correct" in {
+    compare(
+      NodeDef.newBuilder()
+        .setName(s"${getOpName}Test")
+        .setOp(s"$getOpName")
+        .putAttr("T", typeAttr(DataType.DT_FLOAT)),
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/utils/tf/loaders/UnaryOpBase.scala]: for operation with a T, should we define the operation with an extra type D?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Sigmoid.scala (hunk 1 above)
Comment (yangw1234) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Sigmoid.scala]: The original implementation needs a extra forward to fill in the `buffer`.

**Comment by yangw1234** [comment]
pr validation pass

**Comment by yiheng** [comment]
LGTM

**Pull request merged**